### PR TITLE
Gun-shy

### PR DESCRIPTION
**Pull request by Consoleable**
#### Summary
Content "Adds Gun-shy Trait"

#### Purpose of change
Adds a gun-shy trait to character gen. Allows bows and the like but not firearms.

#### Describe the solution
adds gunshy trait, prevents user from firing guns and turret guns, similar functionality to brawler. Allows user to fire primitive ranged weapons like crossbows and bows.

#### Describe alternatives you've considered
Not adding the trait. 

#### Testing
Tested crossbows with and without
Tested bow with and without
Tested glock 17 with and without
Tested firing mode of m16 
Tested trying to fire m249 with and without
Tested firing double barrel shotgun with and without
Tested firing mp 40 with and without
Tested firing 3 inch ordinance rifle with and without
Teated firing staff sling with and without

Guns and ordinance cannot be fired, even if turret
Bows and crossbows and anything with "primitive_ranged_weapon" is allowed as per is_firearm() 

I havent tested every gun but i did one in each category i think its ready for integration.

### CODE REVIEW

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -9990,6 +9990,15 @@
...
     "valid": false
   },
+  {
+    "type": "mutation",
+    "id": "GUNSHY",
+    "name": { "str": "Gun-shy" },
+    "points": -3,
+    "description": "Whether from personal choice or childhood trauma, using a firearm is off-limits to you, even if your life depended on it.",
+    "starting_trait": true,
+    "valid": false
```
Comment (TheShadowFerret) [inline data/json/mutations/mutations.json:10000]:
```suggestion
    "valid": false,
    "cancels": [ "BRAWLER" ]
```

File: data/json/mutations/mutations.json [hunk 2]
```diff
@@ -9990,6 +9990,15 @@
     "starting_trait": true,
     "valid": false
```
Comment (TheShadowFerret) [inline data/json/mutations/mutations.json:9991]:
```suggestion
    "valid": false,
    "cancels": [ "GUNSHY" ]
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]
Since Gunshy encompasses what Brawler does + more, selecting both shouldn't be possible(esp. for people using legacy point select, since that's double points).

**Comment by Consoleable** [comment]
> Since Gunshy encompasses what Brawler does + more, selecting both shouldn't be possible(esp. for people using legacy point select, since that's double points).

This is an excellent suggestion, dunno why i missed that!

**Comment by XygenSS** [comment]
Can you use rail mounted crossbows attached on a firearm?

**Comment by Consoleable** [comment]
> Can you use rail mounted crossbows attached on a firearm?

Will test this after work, most likely not though, still a WIP. 
May have to add support in an additional update depending on how much work it would be to add. 
Good suggestion!

File: data/json/mutations/mutations.json [hunk 3]
```diff
@@ -9988,7 +9988,18 @@
...
     "starting_trait": true,
+    "valid": false,
+    "cancels": [ "GUNSHY" ]
+  },
+  {
+    "type": "mutation",
+    "id": "GUNSHY",
+    "name": { "str": "Gun-Shy" },
+    "points": -3,
+    "description": "Using a firearm is off limits, due to personal philosophy or a traumatic event. You can still use bows and slings.",
```
Comment (Uwuewsky) [inline data/json/mutations/mutations.json:9999]:
```suggestion
    "description": "Using a firearm is off limits, due to personal philosophy or a traumatic event.  You can still use bows and slings.",
```

**Comment by Aimless-dude** [comment]
Will you still be able to use the crossbow turret, the Carroballista?

**Comment by Consoleable** [comment]
> Will you still be able to use the crossbow turret, the Carroballista?

I've been looking into this, and it looks like it's a bit of a challenge to set up. I'm going to make an issue for this and handle it in a followup PR, because addressing this will require changing some more things in turrret.cpp and i don't want to scope creep
The ballistae isn't commonly used or easy to find, so i'll merge this version of the trait for now.

**Pull request merged**